common/scala/src/main/scala/org/apache/openwhisk/core/containerpool/logging/DockerToActivationFileLogStore.scala

```diff
@@ -151,7 +155,10 @@ class DockerToActivationFileLogStore(system: ActorSystem, destinationDirectory:
     logs.runWith(combined)._1.flatMap { seq =>
       val possibleErrors = Set(Messages.logFailure, Messages.truncateLogs(action.limits.logs.asMegaBytes))
       val errored = seq.lastOption.exists(last => possibleErrors.exists(last.contains))
-      val logs = ActivationLogs(seq.toVector)
+      val logs = ActivationLogs(
+        if (activation.response.isContainerError)
+          seq.toVector :+ LogLine(Instant.now.toString, "stderr", Messages.logFailure).toFormattedString
+        else seq.toVector)
```

Comment:
This will write the error log only in this path of the code. You might want to concat it to the stream above so all possible branches get this message, like:

```scala
val logsWithPossibleError = if(activation.response.isContainerError) {
  logs.concat(Source.single(ByteString(LogLine(...))))
} else logs
```

And then work off of that. Note that the `errored` check should be short circuited as well in case there was a ContainerError it should always be true and doesn't need to look at the last message "manually".

Reply:
I agree - @markusthoemmes has a good finding.

Reply:
adapted the code to your proposal